### ISSUE TITLE
`deno lint` needs detection of undefined identifiers (`no-undef`) to be useful

### ISSUE THREAD

**Issue by mitranim** [denoland/deno_lint#1327]
Currently, `deno lint` doesn't complain about missing identifiers. The following JS program passes linting but explodes at runtime:

```js
blah
```

In my experience, this is an _extremely_ common source of bugs. The TS compiler checks this, but not every project uses TS. For JS, a linter able to detect this is essential; as a result, `deno lint` is not sufficient, forcing the use of `eslint`.

The main difficulty seems to be that `no-undef` requires knowledge of allowed globals, with different sets for different environments, and may involve config files like `.eslintrc`, which Deno seems to staunchly oppose. Right now, I don't have a concrete suggestion.
